### PR TITLE
Exception Factory Changes

### PR DESCRIPTION
**Pull request by kpartlow**

### CODE REVIEW

File: src/main/java/com/cedarsoftware/util/io/factory/ThrowableFactory.java [hunk 1]
```diff
@@ -27,24 +29,70 @@
...
+        String msg = (String) jsonObj.get(DETAIL_MESSAGE);
+        JsonObject causeMap = (JsonObject) jsonObj.get(CAUSE);
+
+
+        JsonReader reader = ReaderContext.instance().getReader();
+
+        Throwable cause = causeMap == null ? null : reader.convertParsedMapsToJava(causeMap, Throwable.class);
+        Throwable t = createException(c, msg, cause);
+
+        Object[] stackTraces = (Object[]) jsonObj.get(STACK_TRACE);
```
Comment (jdereg) [inline src/main/java/com/cedarsoftware/util/io/factory/ThrowableFactory.java:47]: What about setting the stackTrace to an empty[]?  I'm thinking from a security perspective, it provides value mainly to a hacker.

**Comment by kpartlow** [comment]
I'm thinking about letting that be configurable..through the Write options.

**Comment by keeldragger** [comment]
happy to see the investment in json-io continues.  it's better than crypto

**Comment by jdereg** [comment]
How the heck are you, keeldragger?  I hope you are having Bunches of fun
piloting your ship around.

**Pull request merged**